File: library/src/main/java/me/shkschneider/skeleton/network/WebServiceException.java
package me.shkschneider.skeleton.network;

import android.support.annotation.IntRange;
import android.support.annotation.NonNull;

import me.shkschneider.skeleton.helper.LocaleHelper;

public class WebServiceException extends Exception {

    public static final int INTERNAL_ERROR = 666;

    private final int mCode;

    WebServiceException(@IntRange(from=0, to=INTERNAL_ERROR) final int responseCode, @NonNull final String responseMessage) {
        super(responseMessage);
        mCode = responseCode;
    }

    public int getCode() {
        return mCode;
    }

    @Override
    public String getMessage() {
        return super.getMessage();
    }

    @Override
    public String toString() {
        return String.format(LocaleHelper.locale(), "%d %s", getCode(), getMessage());
    }

}

File: library/src/main/kotlin/me/shkschneider/skeleton/network/WebServiceException.kt
package me.shkschneider.skeleton.network

import android.support.annotation.IntRange
import java.util.*

class WebServiceException : Exception {

    val code: Int

    constructor(@IntRange(from = 0, to = INTERNAL_ERROR.toLong()) code: Int, message: String? = null) : super(message, null) {
        this.code = code
    }

    override fun toString(): String {
        return String.format(Locale.getDefault(), "%d %s", code, message)
    }

    companion object {

        const val INTERNAL_ERROR = 666

    }

}
